src/Examine.Core/Search/IFacetFullTextField.cs

```diff
@@ -0,0 +1,17 @@
+using Examine.Lucene.Search;
+
+namespace Examine.Search
+{
+    public interface IFacetFullTextField : IFacetField
+    {
+        /// <summary>
+        /// Maximum number of terms to return
+        /// </summary>
+        int MaxCount { get; set; }
```

Comment:
Similarly, this is only ever set by MaxCount() methods, the interface shouldn't expose this as get/set, just get. Sorry for duplicating a lot of these suggestions but I think unless there's an absolute requirement/argument to make any new properties in this PR get/set, they should all just be get and everything should be exposed as immutable in the API.